lib/enigma.rb

```diff
@@ -0,0 +1,63 @@
+class Enigma
+  attr_reader :key,
+              :date
+
+  def initialize
+    @key = nil
+    @date = nil
+  end
+
+  def key_object(key)
```

Comment:
Should this be a responsibility of enigma? Should Key be responsible fo instantiating itself and making the decision what to do with the information given to it?